limit/priority_channel.go

```diff
@@ -1,34 +1,59 @@
 package limit
 
+import (
+	"sync"
+)
+
 // priorityChannel is the channels support the priority of them.
 type priorityChannel struct {
-	Out    chan interface{}
-	High   chan interface{}
-	Normal chan interface{}
-	Low    chan interface{}
-	stopCh chan struct{}
+	Out     chan interface{}
+	High    chan interface{}
+	Normal  chan interface{}
+	Low     chan interface{}
+	closeCh chan struct{}
+
+	refCount uint
+	mu       *sync.Mutex
 }
 
 // initPriorityChannel will create the priorityChannel and initialize it
-func initPriorityChannel(closeCh <-chan struct{}) *priorityChannel {
+func initPriorityChannel() *priorityChannel {
 	pc := priorityChannel{}
 	pc.Out = make(chan interface{})
 	pc.High = make(chan interface{})
 	pc.Normal = make(chan interface{})
 	pc.Low = make(chan interface{})
-	pc.stopCh = make(chan struct{})
+	pc.closeCh = make(chan struct{})
+	pc.mu = new(sync.Mutex)
```

Comment:
https://golang.org/doc/effective_go.html#data
Using zero value is better, instead of handling pointer and initialize with new(), maybe.